### PR TITLE
Feature/complex bottom sheet

### PR DESCRIPTION
**Pull request by daileytj**
Fixes PXBLUE 1280.

#### Changes proposed in this Pull Request:
- Add Complex Bottom Sheet design pattern example

### CODE REVIEW

File: src/app/pages/overlays/complex-bottom-sheet/complex-bottom-sheet.component.html [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        <mat-icon>menu</mat-icon>
+    </button>
+    <div class="mat-title">Complex Bottom Sheet</div>
+    <pxb-spacer></pxb-spacer>
+    <button mat-icon-button (click)="showBottomSheet()">
+        <mat-icon>more_vert</mat-icon>
+    </button>
+</mat-toolbar>
+<mat-list *ngIf="data && data.length > 0">
+    <pxb-info-list-item
```
Comment (joebochill) [inline src/app/pages/overlays/complex-bottom-sheet/complex-bottom-sheet.component.html:12]: Looks like something goofy is happening with the ListItems again (might be box-sizing issue again):
Should look like:
image

Note the icons are vertically centered with the text and left aligned to the edge of the avatar

Currently looks like:

image

File: src/app/pages/overlays/complex-bottom-sheet/complex-bottom-sheet.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    it('should create the service', async(() => {
+        void expect(service).toBeTruthy();
+    }));
+
+    it('updateSort works', () => {
+        service.updateSort(FiltersList.EVENT_TYPE);
+        void expect(service.activeSort).toBe(FiltersList.EVENT_TYPE);
+    });
+
+    it('updateFilters works', () => {
```
Comment (joebochill) [inline src/app/pages/overlays/complex-bottom-sheet/complex-bottom-sheet.component.spec.ts:42]: can we update these test names to match the format `it('should......')`

File: src/app/pages/overlays/complex-bottom-sheet/data.service.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
+/* eslint-disable default-case */
```
Comment (joebochill) [inline src/app/pages/overlays/complex-bottom-sheet/data.service.ts:1]: why are we disabling the default case rule?

File: src/app/pages/overlays/complex-bottom-sheet/data.service.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        }
+    }
+
+    passData(): void {
+        this.dataSource.next(this.sortAlarms(this.filterAlarms(this.fullData)));
+    }
+
+    updateSort(sortType: string): void {
+        // eslint-disable-next-line no-console
+        console.log('data: ', sortType);
```
Comment (joebochill) [inline src/app/pages/overlays/complex-bottom-sheet/data.service.ts:83]: no console logs?

File: src/app/pages/overlays/complex-bottom-sheet/complex-bottom-sheet.component.html (hunk 1 above)
Comment (huayunh) [reply src/app/pages/overlays/complex-bottom-sheet/complex-bottom-sheet.component.html:12]: From a design point of view I personally like the second one better (did I do that in the React mono repo too?)

Comment (daileytj) [reply src/app/pages/overlays/complex-bottom-sheet/complex-bottom-sheet.component.html:12]: For some reason, there was padding on the icons...

Comment (daileytj) [reply src/app/pages/overlays/complex-bottom-sheet/complex-bottom-sheet.component.html:12]: @huayunh The react mono repo example has the icons aligned like the second picture

Comment (joebochill) [reply src/app/pages/overlays/complex-bottom-sheet/complex-bottom-sheet.component.html:12]: The first picture is how the ILI _should_ look (that is, Eric said it). Everything is aligned on the Material keylines. Ian and I have gone back and forth on this a bunch in the past. Center aligning makes it so the icons don't align with anything else in the application. I agree it tends to look nicer in this one case where you have a mix of icons and avatars, but I would advocate for sticking with the left alignment in the majority of cases (and the default). You can achieve the center aligning (as we have done in the React example) by using a transparent avatar background.

**Pull request merged**